languages/scala/reference/README.md

```diff
@@ -0,0 +1,193 @@
+# Scala reference
+
+## Concepts
+
+The Scala concept exercises are based on concepts. The list below contains the concepts that have been identified for the Scala language.
+
+### Language-unique
+
+- Pattern matching
+- Functional Programming
+- Immutability
+
+### Functional
+
+- [Functions][functions]
+  - [Pure functions][pure_functions]
+  - [Higher-order functions][higher_order_functions]
+  - [Anonymous functions][anonymous_functions]
+  - [Nested functions][nested_functions]
+  - [Function composition][function_composition]
+  - [Partial application][partial_application]
+  - [Recursion][recursion]
+    - Tail recursion
+  - Currying
+  - Implicit returns
+  - Type annotations
+- [Immutability][immutability]
+- [Pattern matching][pattern_matching]
+  - Discards
+  - Guard clauses
+- [Type inference][type_inference]
+  - Automatic generalisation
+- [Pipelines][pipelines]
+- [Expression-oriented (not statement oriented)][expression_oriented]
+
+### Object-oriented
+
+The core OO features an Scala developer should know about are:
+
+- [Classes][classes]
+- [case class][case class]
+- [Encapsulation][encapsulation]
+- [Objects][objects]
+- [Interfaces][interfaces]
+  - Object expressions
+- [Methods][methods]
+  - Constructors
+  - Apply
+  - Unapply
+  - Method arguments
+    - Named arguments
+    - [Optional arguments][optional_arguments]
+    - Out parameters
+- Fields
+- Properties
+- Type extensions
+- Implicit class
+
+While Scala has OO features, it is a "functional first" language. It is important to know how and when to apply the OO features, and which OO features to use.
+
+### General
+
+- Values
+  - Assignment
+- Scoping
+  - Namespaces
+  - Modules
+    - Extending modules
+  - Imports (`import`)
+  - Visibility (`public`, `private`, etc.)
+- Numbers
+  - Math operators
+  - Bitwise manipulation
+  - Arithmetic overflow
+- Comparison
+  - [Equality][equality] (`Equals`, `GetHashCode`)
+  - Ordering
+- Conditionals
+  - Boolean logic
+  - [Conditionals: if][conditionals]
+  - Conditionals: while
+- Enumeration
+  - [Enumeration: for loop][enumeration]
+- Collections
+  - [Generics][generics]
+  - Sequence/for expressions
+  - Iterators (yield)
+  - Collections: mapping
+  - Collections: filtering
+  - Collections: reducing
+  - Collections: ordering
+  - Collections: combining
+  - Ranges
+    - Slicing
+- Resources
+  - Resource passing (by reference/by value)
+- Conversions
+  - Implicit
+  - Explicit
+- Concurrency
+  - Concurrent collections `par`
+  - Future
+- Exceptions
+  - `Try`
+  - `Either`
+- Code quotations
+- Attributes
+- Nullability
+  - `Option`
+- String formatting
+  - Formatting types
+
+### Types
+
+- [Strings][string]
+- [Characters][char]
+- [Booleans][bool]
+- [Option][option]
+- [Either][either]
+- [Try][try]
+- [Future][future]
+- [Tuples][tuple]
+- Unit
+- Numbers
```

Comment:
I think adding `BigInteger` or `BigDecimal` is a good idea. Either of them should be sufficient.

Reply:
Added, I though those were part of the java library tho ?

Reply:
Is it a data type that people commonly use in Scala?

Reply:
> […] I thought those were part of the java library tho ?

There is a Scala version of BigDecimals and BigIntegers. Scala uses the Java ones under the hood, but the interface is a bit different (from what I can tell). 

> Is it a data type that people commonly use in Scala?

IMO: It's not commonly used, but it should be used way more often because there are use cases where `float` and `double` fall short.

Reply:
> There is a Scala version of BigDecimals and BigIntegers.

This seems to indicate that inclusion is warranted.

Reply:
I'll consider this resolved then :)